crates/orderbook/src/database/orders.rs

```diff
@@ -75,6 +77,7 @@ pub trait OrderStoring: Send + Sync {
         offset: u64,
         limit: Option<u64>,
     ) -> Result<Vec<Order>>;
+    async fn order_status(&self, order_uid: &OrderUid) -> Result<Status>;
```

Comment:
shouldn't return `Result<Option<Status>>` what if the order doesn't exist?

Reply:
In that case, it returns an error because this API is not expected to check whether the order exists.

Reply:
But it is an inconsistent pattern. See this [function](https://github.com/cowprotocol/services/pull/2804/files#diff-d22c8612ee861d55e838fdc5487a1c139da73b795295670acdcccea91f805011R71).

Reply:
IMO, semantic of `single_order` and `order_status` is different. The former fetches the order if it exists and latter should return the order's status and absence of it should be considered as an error.

Reply:
I think returning an error makes sense.
I'd argue the error should be 404 in this case, though.

Reply:
It returns 404 now.